MaterialLib/MPL/CreatePhase.cpp

```diff
@@ -0,0 +1,112 @@
+/**
+ * \file
+ * \author Norbert Grunwald
+ * \date   07.09.2017
+ *
+ * \copyright
+ * Copyright (c) 2012-2019, OpenGeoSys Community (http://www.opengeosys.org)
+ *            Distributed under a Modified BSD License.
+ *              See accompanying file LICENSE.txt or
+ *              http://www.opengeosys.org/project/license
+ *
+ */
+
+#include "CreatePhase.h"
+
+#include <set>
+#include <string>
+
+#include "BaseLib/ConfigTree.h"
+
+#include "CreateComponent.h"
+#include "CreateProperty.h"
+#include "Phase.h"
+
+namespace
+{
+std::unique_ptr<MaterialPropertyLib::Phase> createPhase(
+    BaseLib::ConfigTree const& config)
+{
+    using namespace MaterialPropertyLib;
+
+    //! \ogs_file_param{prj__media__medium__phases__phase__type}
+    auto&& phase_type = config.getConfigParameter<std::string>("type");
+
+    if (phase_type.empty())
+    {
+        OGS_FATAL("Phase type is a mandatory field and cannot be empty.");
+    }
+
+    std::array<std::string, 4> const allowed_phase_types = {
+        {"Solid", "AqueousLiquid", "NonAqueousLiquid", "Gas"}};
+
+    if (std::none_of(allowed_phase_types.begin(),
+                     allowed_phase_types.end(),
+                     [&phase_type](std::string const& type) {
+                         return phase_type == type;
+                     }))
+    {
+        ERR("Phase type should be one of:");
+        for (auto const type : allowed_phase_types)
+        {
+            ERR(type.c_str());
```

Comment:
Is this valid? I thought it must be ```ERR("%s", type.c_str())```.

Reply:
It's just a char array, with substitution or without.